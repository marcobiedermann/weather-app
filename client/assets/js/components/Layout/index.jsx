import PropTypes from 'prop-types';
import React from 'react';
import Footer from '../Footer';
import Header from '../Header';
import Main from '../Main';
import Navigation from '../Navigation';
import * as routes from '../../constants/routes';

const Layout = (props) => {
  const { children } = props;

  return (
    <div>
      <Header>
        <Navigation
          routes={[
            routes.INDEX,
            routes.CITIES,
            routes.SETTINGS,
          ]}
        />
      </Header>
      <Main>{children}</Main>
      <Footer>FOOTER</Footer>
    </div>
  );
};

Layout.propTypes = {
  children: PropTypes.node,
};

Layout.defaultProps = {
  children: null,
};

export default Layout;
