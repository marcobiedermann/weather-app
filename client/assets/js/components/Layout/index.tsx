import PropTypes from 'prop-types';
import React from 'react';
import Footer from '../Footer';
import Grid from '../Grid';
import Header from '../Header';
import Main from '../Main';
import Navigation from '../Navigation';
import * as routes from '../../constants/routes';
import './style.module.css';

const Layout = (props) => {
  const { children, ...otherProps } = props;

  return (
    <div {...otherProps}>
      <Header>
        <Grid>
          <Navigation routes={[routes.INDEX, routes.CITIES, routes.SEARCH, routes.SETTINGS]} />
        </Grid>
      </Header>
      <Main>
        <Grid>{children}</Grid>
      </Main>
      <Footer>
        <Grid>FOOTER</Grid>
      </Footer>
    </div>
  );
};

Layout.propTypes = {
  children: PropTypes.node,
};

Layout.defaultProps = {
  children: null,
};

export default Layout;
