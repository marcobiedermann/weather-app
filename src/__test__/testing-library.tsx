import { RenderOptions, render } from '@testing-library/react';
import { ReactElement, ReactNode } from 'react';
import { I18nextProvider } from 'react-i18next';
import { MemoryRouter } from 'react-router-dom';
import i18n from './i18n';

interface ProvidersProps {
  children: ReactNode;
}

function Providers(props: ProvidersProps) {
  const { children } = props;

  return (
    <I18nextProvider i18n={i18n}>
      <MemoryRouter>{children}</MemoryRouter>
    </I18nextProvider>
  );
}

function customRender(ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) {
  return render(ui, {
    wrapper: Providers,
    ...options,
  });
}

export { customRender as render };
