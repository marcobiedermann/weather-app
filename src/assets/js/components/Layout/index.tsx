import React, { FC } from 'react';
import * as routes from '../../constants/routes';
import Footer from '../Footer';
import Grid from '../Grid';
import Header from '../Header';
import Main from '../Main';
import Navigation from '../Navigation';
import styles from './style.module.css';

const Layout: FC = (props) => {
  const { children } = props;

  return (
    <div className={styles.layout}>
      <Header>
        <Grid>
          <Navigation routes={[routes.INDEX, routes.CITIES, routes.SEARCH, routes.SETTINGS]} />
        </Grid>
      </Header>
      <Main>
        <Grid>{children}</Grid>
      </Main>
      <Footer>
        <Grid>FOOTER</Grid>
      </Footer>
    </div>
  );
};

export default Layout;
