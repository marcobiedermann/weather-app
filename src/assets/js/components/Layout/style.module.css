* {
  box-sizing: inherit;
}

html {
  box-sizing: border-box;
}

body {
  background-color: #fcfcfc;
  color: #fcfcfc;
  display: grid;
  font-family: 'Helvetica Neue', sans-serif;
  line-height: 1.5;
  margin: 0;
  min-height: 100vh;
  place-items: center;
}

a {
  color: inherit;
  text-decoration: none;
}

button,
input,
select,
textarea {
  border: 0;
  font: inherit;
  margin: 0;
  padding: 0;
}

button {
  cursor: pointer;
}

h1,
h2,
h3 {
  font-weight: 500;
}

table {
  width: 100%;
}

.layout {
  background-image: linear-gradient(#b4c4d1, #505674);
  border-radius: 1.5em;

  /* box-shadow:
    0 2.8px 2.2px rgba(0, 0, 0, 0.02),
    0 6.7px 5.3px rgba(0, 0, 0, 0.028),
    0 12.5px 10px rgba(0, 0, 0, 0.035),
    0 22.3px 17.9px rgba(0, 0, 0, 0.042),
    0 41.8px 33.4px rgba(0, 0, 0, 0.05),
    0 100px 80px rgba(0, 0, 0, 0.07); */
  display: flex;
  flex-direction: column;
  height: 853px;
  padding: 1.5em;
  width: 480px;
}
