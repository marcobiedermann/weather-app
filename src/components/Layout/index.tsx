import { ReactNode } from 'react';
import Footer from '../Footer';
import Header from '../Header';
import Main from '../Main';
import Navigation from '../Navigation';
import styles from './style.module.css';
import { Grid, Settings } from 'react-feather';
import Search from '../Search';
import { createSearchParams, useNavigate } from 'react-router-dom';

interface LayoutProps {
  children: ReactNode;
}

function Layout(props: LayoutProps): JSX.Element {
  const { children } = props;
  const navigate = useNavigate();

  return (
    <div className={styles.layout}>
      <div
        style={{
          display: 'flex',
          flexDirection: 'column',
          flexGrow: 1,
        }}
      >
        <Header>
          <Search
            onSubmit={(data) => {
              const { query } = data;
              const search = createSearchParams({
                query,
              }).toString();

              navigate({
                pathname: 'search',
                search,
              });
            }}
          />
        </Header>
        <Main>{children}</Main>
        <Footer>
          <p>
            Data provided by the{' '}
            <a href="https://openweathermap.org/" rel="noopener noreferrer" target="_blank">
              OpenWeather
            </a>{' '}
            API.
          </p>
        </Footer>
      </div>
      <aside
        style={{
          backgroundColor: '#eef2f3',
          color: '#0e2546',
          display: 'flex',
          flexDirection: 'column',
          justifyContent: 'space-between',
          minWidth: 300,
          order: -1,
          paddingBlock: '3em',
        }}
      >
        <div>
          <div style={{ paddingInline: '3em' }}>
            <h4>Application</h4>
          </div>
          <Navigation
            routes={[
              {
                name: 'Dashboard',
                path: '/cities',
                icon: <Grid />,
              },
            ]}
          />
        </div>
        <div>
          <div style={{ paddingInline: '3em' }}>
            <h4>System</h4>
          </div>
          <Navigation
            routes={[
              {
                name: 'Settings',
                path: '/settings',
                icon: <Settings />,
              },
            ]}
          />
        </div>
      </aside>
    </div>
  );
}

export default Layout;
