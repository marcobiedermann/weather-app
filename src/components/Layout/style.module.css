* {
  box-sizing: inherit;
}

html {
  box-sizing: border-box;
}

body {
  display: grid;
  font-family: Figtree, sans-serif;
  line-height: 1.5;
  margin: 0;
  min-height: 100vh;
}

a {
  color: inherit;
}

button,
input,
select,
textarea {
  background: none;
  border: 0;
  font: inherit;
  margin: 0;
  padding: 0;
}

button {
  cursor: pointer;
}

h1,
h2,
h3,
h4,
h5 {
  color: #0e2546;
  font-weight: 600;
}

h4 {
  margin: 0;
}

h5 {
  margin: 0;
}

label {
  display: inline-block;
  font-weight: 600;
  margin-bottom: 0.5em;
}

p {
  margin-block: 0.5em 1.5em;
}

select {
  background-color: #fcfcfc;
  border-radius: 0.25em;
  padding: 1em 1.5em;
}

section {
  padding-block: 2em;
}

table {
  width: 100%;
}

.layout {
  display: flex;
  min-block-size: 100vh;
}
