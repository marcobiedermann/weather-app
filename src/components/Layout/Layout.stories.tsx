import type { Meta, StoryObj } from '@storybook/react';
import Layout from '.';

const meta: Meta<typeof Layout> = {
  component: Layout,
  args: {
    children: 'Content',
  },
};

type Story = StoryObj<typeof Layout>;

const Default: Story = {};

export { Default };
export default meta;
