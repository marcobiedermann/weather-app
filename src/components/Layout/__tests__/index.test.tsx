import { describe, expect, it } from 'vitest';
import Layout from '..';
import { render } from '../../../__test__/testing-library';

describe('components/Layout', () => {
  it('renders correctly', () => {
    const { asFragment } = render(<Layout>Content</Layout>);

    expect(asFragment()).toMatchSnapshot();
  });
});
