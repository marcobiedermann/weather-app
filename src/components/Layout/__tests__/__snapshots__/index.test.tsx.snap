// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Layout renders correctly 1`] = `
<DocumentFragment>
  <div
    class="layout"
  >
    <header>
      <div
        class="grid"
      >
        <nav
          class="navigation"
        >
          <ul
            class="navigation__list"
          >
            <li
              class="navigation__item"
            >
              <a
                class="navigation__link"
                href="/cities"
              >
                Cities
              </a>
            </li>
            <li
              class="navigation__item"
            >
              <a
                class="navigation__link"
                href="/search"
              >
                Search
              </a>
            </li>
            <li
              class="navigation__item"
            >
              <a
                class="navigation__link"
                href="/settings"
              >
                Settings
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
    <main
      class="main"
    >
      <div
        class="grid"
      >
        Content
      </div>
    </main>
    <div>
      <div
        class="grid"
      >
        <p>
          <a
            href="https://openweathermap.org/"
            rel="noopener noreferrer"
            target="_blank"
          >
            OpenWeather
          </a>
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
